docs/data/date-pickers/localization/LocalizedDatePicker.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import frLocale from 'date-fns/locale/fr';
+import ruLocale from 'date-fns/locale/ru';
+import deLocale from 'date-fns/locale/de';
+import enLocale from 'date-fns/locale/en-US';
+import ToggleButton from '@mui/material/ToggleButton';
+import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
+import TextField from '@mui/material/TextField';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+
+const localeMap = {
+  en: enLocale,
+  fr: frLocale,
+  ru: ruLocale,
+  de: deLocale,
+};
+
+const maskMap = {
+  fr: '__/__/____',
+  en: '__/__/____',
+  ru: '__.__.____',
+  de: '__.__.____',
+};
+
+export default function LocalizedDatePicker() {
+  const [locale, setLocale] = React.useState<keyof typeof maskMap>('ru');
+  const [value, setValue] = React.useState<Date | null>(new Date());
+
+  const selectLocale = (newLocale: any) => {
+    setLocale(newLocale);
+  };
+
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns} locale={localeMap[locale]}>
```

Comment:
Should we name it more explicitly to avoid any confusion with the `localeText` ? `dateLocale` for instance ?

Reply:
What about `adapterLocale` because it is the argument passed to the adapter

Reply:
Works for me :+1: